libs/ui/src/ui_strings/with_audio.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import getDeepValue from 'lodash.get';
+
+import { LanguageCode } from '@votingworks/types';
+
+import { useAudioContext } from './audio_context';
+import { useCurrentLanguage } from '../hooks/use_current_language';
+import { UiStringsReactQueryApi } from '../hooks/ui_strings_api';
+
+export enum UiStringAudioDataAttributeName {
+  I18N_KEY = 'data-i18n-key',
+  LANGUAGE_CODE = 'data-language-code',
+}
+
+export interface WithAudioProps {
+  children: React.ReactNode;
+  i18nKey: string;
+}
+
+/**
+ * Utility component for pre-fetching audio clip data to help shorten audio
+ * playback response times.
+ */
+function PrefetchAudioClip(props: {
+  api: UiStringsReactQueryApi;
+  audioId: string;
+  languageCode: LanguageCode;
+}) {
+  const { api, audioId, languageCode } = props;
+
+  api.getAudioClip.useQuery({
+    id: audioId,
+    languageCode,
+  });
+
+  return null;
+}
+
+/**
+ * Utility component for pre-fetching audio clip data to help shorten audio
+ * playback response times.
+ */
+function PrefetchAudioClips(props: {
+  api: UiStringsReactQueryApi;
+  i18nKey: string;
+  languageCode: LanguageCode;
+}) {
+  const { api, i18nKey, languageCode } = props;
+
+  const audioIdsQuery = api.getAudioIds.useQuery(languageCode);
+  const audioIdMappings = audioIdsQuery.isFetching
+    ? undefined
+    : audioIdsQuery.data;
```

Comment:
I know in most places, we check `isSuccess` instead of `isFetching`, e.g.

```ts
if (!audioIdsQuery.isSuccess) {
  return null;
}

const audioIdMappings = audioIdsQuery.data;
```

Is there a particular reason to use `isFetching` instead here?

Reply:
Ohh, good question -- wrote this a while ago and now don't remember why 😂 
The `isSuccess` guard pattern's definitely the right way to go here - thanks